replica/table.cc

```diff
@@ -803,17 +862,14 @@ table::stop() {
     }
     return _async_gate.close().then([this] {
         return await_pending_ops().finally([this] {
-            return _memtables->flush().finally([this] {
-                return _compaction_manager.remove(as_table_state()).then([this] {
-                    return _sstable_deletion_gate.close().then([this] {
-                        return get_row_cache().invalidate(row_cache::external_updater([this] {
-                            _main_sstables = _compaction_strategy.make_sstable_set(_schema);
-                            _maintenance_sstables = make_maintenance_sstable_set();
-                            _sstables = make_compound_sstable_set();
-                        })).then([this] {
-                            _compaction_strategy.get_backlog_tracker().disable();
-                            _cache.refresh_snapshot();
-                        });
+            return _compaction_group->stop().then([this] {
+                return _sstable_deletion_gate.close().then([this] {
+                    return get_row_cache().invalidate(row_cache::external_updater([this] {
+                        _compaction_group->clear_sstables();
+                        _sstables = make_compound_sstable_set();
+                    })).then([this] {
+                        _compaction_strategy.get_backlog_tracker().disable();
+                        _cache.refresh_snapshot();
```

Comment:
nit: I think that if make this a coroutine as a preliminary change, it will be more obvious what exactly changed by reducing the churn and leaving only the parts still relevant to table and what moves to compaction_group.

Reply:
In the first patch which touches table::stop(), I used that technique of breaking indentation to make it easy to verify what changed, then restored indentation in a later patch. So I assume that making it a couroutine is not needed for this specific reason. But coroutine has its benefits so I will convert table::stop() into coroutine in preliminary patch.